tests/integration/_experimental/dataset/test_dataset.py

```diff
@@ -148,17 +149,18 @@ def test__register_dataset__composite() -> None:
         for i in range(1, 20)
     ]
     columns = datapoints[0].keys()
+    id_fields = ["id"]
 
     df = pd.DataFrame(datapoints[:10], columns=columns)
-    register_dataset(name, df)
+    register_dataset(name, df, id_fields=id_fields)
 
     loaded_datapoints = fetch_dataset(name)
     loaded_datapoints = loaded_datapoints.sort_values("total_word_count", ignore_index=True).reindex(columns=columns)
     assert_frame_equal(df, loaded_datapoints)
 
     # update dataset
     df = pd.DataFrame(datapoints[:5] + datapoints[7:15], columns=columns)
-    register_dataset(name, df)
+    register_dataset(name, df, id_fields=id_fields)
```

Comment:
Not sure if we should do it in this PR, but I think we should definitely add some tests to cover [this logic](https://github.com/kolenaIO/kolena/blob/ab94b07b0951d104d78fa70ff60c64c8e518e39f/kolena/_experimental/dataset/_dataset.py#L221-L227)

Reply:
added unit tests